components/definition/cellRules/sections/mechanisms/NetPyNEMechanismThumbnail.js

```diff
@@ -50,26 +48,33 @@ export default class NetPyNEMechanismThumbnail extends React.Component {
   };
 
   render() {
+    const { name, selected } = this.props;
+    const { dialogOpen, isHovered } = this.state;
     return (
-      <div style={{float: "left"}}>
-      <IconButton
-        id={"mechThumb"+this.props.name}
-        onMouseEnter={this.handleHoverIn}
-        onMouseLeave={this.handleHoverOut}
-        className={"gearThumbButton " + (this.props.selected ? "selectedGearButton" : "")}
-        onClick={this.handleClick}>
+      <div style={styles.main}>
+        <IconButton
+          id={`mechThumb${name}`}
+          onMouseEnter={ () => this.setState({isHovered: true}) }
+          onMouseLeave={ () => this.setState({isHovered: false}) }
+          data-tooltip={isHovered && name.length > 10 ? name : undefined}
+          className={"gearThumbButton " + (selected ? "selectedGearButton" : "")}
+          onClick={ (e, v) => this.handleClick(v) }
+        >
           <div>
-           {(this.state.isHovered && this.props.selected) ? 
-            <FontIcon className="fa fa-trash-o" color="white" hoverColor="white"  style={{zIndex:10,marginTop:37,marginLeft:40,position:"absolute",backgroundColor:"#f23d7a"}}/> :
-            <span className={"gearThumbLabel"}> {this.props.name}</span>}
-           <FontIcon color={changeColor} hoverColor={hoverColor} className="gpt-fullgear"/>
+            {(isHovered && selected) 
```

Comment:
Can we change this to a classic if/else? Probably you can even extract it to a method and maybe reuse in several places?